sklearn/linear_model/tests/test_base.py

```diff
@@ -72,7 +72,7 @@ def test_linear_regression_sample_weights(
     sample_weight = 1.0 + rng.uniform(size=n_samples)
 
     # LinearRegression with explicit sample_weight
-    reg = LinearRegression(fit_intercept=fit_intercept)
+    reg = LinearRegression(fit_intercept=fit_intercept, tol=1e-16)
```

Comment:
So I guess this is not needed if we go for backward-compat i.e. `tol=1e-6` by default?
```suggestion
    reg = LinearRegression(fit_intercept=fit_intercept)
```

Reply:
I would keep it anyway because it's kind of "lucky" that it works with a higher tol. The reason tol was added was because it made a sample weight test fail and this test is also checking sample weights. Those tests require good convergence.

Reply:
OK fair enough!